src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -212,19 +161,106 @@ object ResolverUtils {
   private def getColumnNameFromSchema(
       schema: StructType,
       resolvedColNameParts: Seq[String],
-      resolver: Resolver): Seq[String] = resolvedColNameParts match {
+      resolver: Resolver,
+      throwIfNotInSchema: Boolean = true): Seq[String] = resolvedColNameParts match {
     case h :: tail =>
-      val field = schema.find(f => resolver(f.name, h)).get
-      field match {
-        case StructField(name, s: StructType, _, _) =>
-          name +: getColumnNameFromSchema(s, tail, resolver)
-        case StructField(_, _: ArrayType, _, _) =>
-          // TODO: Nested arrays will be supported later
-          throw HyperspaceException("Array types are not supported.")
-        case StructField(_, _: MapType, _, _) =>
-          // TODO: Nested maps will be supported later
-          throw HyperspaceException("Map types are not supported")
-        case f => Seq(f.name)
+      val fieldOpt = schema.find(f => resolver(f.name, h))
+      fieldOpt match {
+        case Some(field) =>
+          field match {
+            case StructField(name, s: StructType, _, _) =>
+              name +: getColumnNameFromSchema(s, tail, resolver)
+            case StructField(_, _: ArrayType, _, _) =>
+              // TODO: Nested arrays will be supported later
+              throw HyperspaceException("Array types are not supported.")
+            case StructField(_, _: MapType, _, _) =>
+              // TODO: Nested maps will be supported later
+              throw HyperspaceException("Map types are not supported")
+            case f => Seq(f.name)
+          }
+        case _ =>
+          if (throwIfNotInSchema) {
+            throw HyperspaceException(s"Hyperspace cannot not find $h in schema")
+          } else {
+            if (tail.nonEmpty) {
+              h +: getColumnNameFromSchema(schema, tail, resolver, throwIfNotInSchema)
+            } else {
+              Seq(h)
+            }
+          }
+      }
+
+  }
+
+  /**
+   * [[ResolvedColumn]] stores information when a column name is resolved against the
+   * analyzed plan and its schema.
+   *
+   * Outside unit tests, this object should not be created directly, but via the `resolve` function,
+   * or `ResolvedColumn.apply` with a normalized name.
+   *
+   * @param name The column name resolved from an analyzed plan.
+   * @param isNested Flag to denote if this column is nested or not.
+   */
+  private[hyperspace] case class ResolvedColumn(name: String, isNested: Boolean) {
```

Comment:
Could you move this function up to reduce diff?

Reply:
I will but this is the way `scalafmt` optimises the code.